pyls_spyder/plugin.py

```diff
@@ -17,8 +17,30 @@
 from pyls.config.config import Config
 from pyls.workspace import Workspace, Document
 
-CELL_REGEX = re.compile(r'^[\t ]*# (%%+)(.*)?$')
-BLOCK_REGEX = re.compile(r'^[\t ]*# (--+)(.*)?$')
+# Local imports
+from pyls_spyder.utils import RegexEvaluator
+
+# Code cell regular expressions
+CELL_PERCENTAGE, CELL_PERCENTAGE_REGEX = (
+    'CELL_PERCENTAGE', re.compile(r'^[\t ]*# ?(%%+)(.*)?$'))
+CELL_CODECELL, CELL_CODECELL_REGEX = (
+    'CELL_CODECELL', re.compile(r'^[\t ]*# ?(<codecell>)(.*)?$'))
+CELL_IN, CELL_IN_REGEX = (
+    'CELL_IN', re.compile(r'^[\t ]*# ?(In\[)([^\]\r\n]*)?\]?$'))
```

Comment:
Please add a comment above all these constants to mention what kind of cell each pattern detects.